Mage.Sets/src/mage/cards/k/KjeldoranEliteGuard.java

```diff
@@ -0,0 +1,128 @@
+import mage.abilities.decorator.ConditionalActivatedAbility;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.SacrificeTargetEffect;
+import mage.abilities.effects.common.continuous.BoostTargetEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+import mage.target.targetpointer.FixedTarget;
+
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public class KjeldoranEliteGuard extends CardImpl {
+
+    public KjeldoranEliteGuard(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{W}");
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Target creature gets +2/+2 until end of turn.
+        // When that creature leaves the battlefield this turn, sacrifice Kjeldoran Elite Guard.
+        // Activate only during combat.
+        Ability ability = new ConditionalActivatedAbility(
+                Zone.BATTLEFIELD,
+                new KjeldoranEliteGuardEffect(),
+                new TapSourceCost(),
+                new IsPhaseCondition(TurnPhase.COMBAT, false));
+        ability.addTarget(new TargetControlledCreaturePermanent());
+        this.addAbility(ability);
+    }
+
+    private KjeldoranEliteGuard(final KjeldoranEliteGuard card) { super(card); }
+
+    @Override
+    public Card copy() {
+        return new KjeldoranEliteGuard(this);
+    }
+}
+
+class KjeldoranEliteGuardEffect extends OneShotEffect {
+
+    KjeldoranEliteGuardEffect() {
+        super(Outcome.Neutral);
+        staticText = "Target creature gets +2/+2 until end of turn. "
+                + "When that creature leaves the battlefield this turn, sacrifice Kjeldoran Elite Guard. "
+                + "Activate only during combat.";
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (source == null) { return false; }
+
+        Permanent targetCreature = game.getPermanent(source.getFirstTarget());
+        if (targetCreature == null) { return false; }
+
+        // Target creature gets +2/+2 until end of turn.
+        BoostTargetEffect buffEffect = new BoostTargetEffect(2, 2, Duration.EndOfTurn);
+        buffEffect.setTargetPointer(new FixedTarget(source.getFirstTarget(), game));
+        game.addEffect(buffEffect, source);
+
+        // When that creature leaves the battlefield this turn, sacrifice Kjeldoran Elite Guard.
+        SacrificeTargetEffect sacrificeKjeldoranEliteGuardEffect = new SacrificeTargetEffect();
+        sacrificeKjeldoranEliteGuardEffect.setTargetPointer(new FixedTarget(source.getSourceId(), game));
```

Comment:
you can just use `SacrificeSourceEffect` instead of setting a target, which also means you don't have to worry about whether the source permanent is even still on the battlefield